lib/ClangImporter/IAMInference.cpp

```diff
@@ -165,6 +166,32 @@ static unsigned dropWordUniq(StringRef str, StringRef word, NameBuffer &out) {
   return numDropped;
 }
 
+static clang::Module *getSubmodule(const clang::NamedDecl *decl, clang::Sema &clangSema) {
```

Comment:
You're right, and I'm guilty of one of them. I think it would make sense as at the very least as a static method on the Impl, and more reasonably part of a refactoring of Clang-specific isolated reasoning.